app/Console/Commands/AdsProcessTx.php

```diff
@@ -0,0 +1,229 @@
+use Adshares\Adserver\Facades\DB;
+use Adshares\Adserver\Models\AdsTxIn;
+use Adshares\Adserver\Models\User;
+use Adshares\Adserver\Models\UserLedger;
+use Illuminate\Console\Command;
+
+class AdsProcessTx extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'ads:process-tx';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Processes incoming txs';
+
+    /**
+     * Command ended without error
+     */
+    const EXIT_CODE_SUCCESS = 0;
+
+    /**
+     * Command ended prematurely because block ids could not be updated
+     */
+    const EXIT_CODE_CANNOT_GET_BLOCK_IDS = 1;
+
+    /**
+     * @var string blockchain address of AdServer
+     */
+    private $adServerAddress;
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->adServerAddress = config('app.adshares_address');
+    }
+
+    /**
+     * Execute the console command.
+     * @param AdsClient $adsClient
+     * @return int
+     */
+    public function handle(AdsClient $adsClient)
+    {
+        try {
+            $this->updateBlockIds($adsClient);
+        } catch (CommandException $exc) {
+            $code = $exc->getCode();
+            $message = $exc->getMessage();
+            $this->error("Cannot update blocks due to CommandException:\n"
+                . "Code:\n  ${code}\n"
+                . "Message:\n  ${message}\n");
+            return self::EXIT_CODE_CANNOT_GET_BLOCK_IDS;
+        }
+
+        $dbTxs = AdsTxIn::where('status', AdsTxIn::STATUS_NEW)->get();
+
+        foreach ($dbTxs as $dbTx) {
+            $this->handleDbTx($adsClient, $dbTx);
+        }
+        return self::EXIT_CODE_SUCCESS;
+    }
+
+    /**
+     * Extracts uuid from tx message.
+     * @param string $message tx message
+     * @return string uuid as hex string
+     */
+    private function extractUuidFromMessage(string $message): string
+    {
+        return substr($message, -32);
+    }
+
+    /**
+     * Updates block data.
+     * @param AdsClient $adsClient
+     * @throws CommandException
+     */
+    private function updateBlockIds(AdsClient $adsClient)
+    {
+        $attempt = 0;
+        $attemptMax = 5;
+        while (true) {
+            try {
+                $response = $adsClient->getBlockIds();
+                $updatedBlocks = $response->getUpdatedBlocks();
+                $this->info("Updated blocks: ${updatedBlocks}");
+                if ($updatedBlocks === 0) {
+                    break;
+                }
+            } catch (CommandException $exc) {
+                if (CommandError::GET_SIGNATURE_UNAVAILABLE === $exc->getCode()
+                    && ++$attempt < $attemptMax) {
+                    // try again after 3 seconds sleep
+                    sleep(3);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this call to "sleep". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS2964)